packages/core-mobile/app/store/notifications/listeners/listeners.ts

```diff
@@ -55,4 +71,36 @@ export const addNotificationsListeners = (
     actionCreator: onAppUnlocked,
     effect: scheduleNotificationsForActiveStakesPeriodically
   })
+
+  startListening({
+    matcher: isAnyOf(
+      onRehydrationComplete,
+      setAccounts,
+      setNonActiveAccounts,
+      setAccount,
+      onLogOut,
+      turnOnNotificationsFor,
+      turnOffNotificationsFor
+    ),
+    effect: async (action, listenerApi) =>
+      await setupBalanceChangeNotifications(listenerApi, action).catch(
```

Comment:
can we separate out the subscribe and unsubscribe logic? I don't think it's a good idea to mix the 2 together. ideally, we should only listen to 1 or 2 actions for each logic. this list is kinda a lot to manage.

Reply:
changed in [Split logic of subscribing to bal change notifications](https://github.com/ava-labs/core-mobile/pull/1659/commits/2d503d6f25c5cbb856afcee77b0e6119268fd8cb)

Reply:
[Split logic of subscribing to bal change notifications](https://github.com/ava-labs/core-mobile/pull/1659/commits/2d503d6f25c5cbb856afcee77b0e6119268fd8cb)